Files/Filesystem/DriveItem.cs

```diff
@@ -49,20 +49,27 @@ public DriveItem(StorageFolder root, DriveType type)
                 return await root.Properties.RetrievePropertiesAsync(new[] { "System.FreeSpace", "System.Capacity" });
             }).Result;
 
-            try
+            if (properties.ContainsKey("System.Capacity") && properties.ContainsKey("System.FreeSpace"))
             {
-                MaxSpace = ByteSize.FromBytes((ulong)properties["System.Capacity"]);
-                FreeSpace = ByteSize.FromBytes((ulong)properties["System.FreeSpace"]);
-
-                SpaceUsed = MaxSpace - FreeSpace;
-                SpaceText = string.Format(
-                    ResourceController.GetTranslation("DriveFreeSpaceAndCapacity"),
-                    FreeSpace.ToBinaryString().ConvertSizeAbbreviation(),
-                    MaxSpace.ToBinaryString().ConvertSizeAbbreviation());
+                try
+                {
+                    MaxSpace = ByteSize.FromBytes((ulong)properties["System.Capacity"]);
+                    FreeSpace = ByteSize.FromBytes((ulong)properties["System.FreeSpace"]);
+
+                    SpaceUsed = MaxSpace - FreeSpace;
+                    SpaceText = string.Format(
+                        ResourceController.GetTranslation("DriveFreeSpaceAndCapacity"),
+                        FreeSpace.ToBinaryString().ConvertSizeAbbreviation(),
+                        MaxSpace.ToBinaryString().ConvertSizeAbbreviation());
+                }
+                catch (NullReferenceException)
+                {
+                    SpaceText = ResourceController.GetTranslation("DriveCapacityUnknown");
+                }
```

Comment:
Are these changes necessary for this pull request?

Reply:
Yes because actually two exceptions are thrown for certain drives which causes a error notification on startup.